draft-ietf-gnap-core-protocol.md

```diff
@@ -5365,3 +5365,693 @@ justification for the data type representing the same basic kind of thing it ext
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "iss-sub"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "iss": {
+                "type": "string"
+              },
+              "sub": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "uri",
+              "sub"
+            ]
+          }
+        }
+      ]
+    },
+    "access_token": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "string"
+        },
+        "manage": {
+          "type": "string"
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/resource"
+          }
+        },
+        "expires_in": {
+          "type": "integer"
+        },
+        "key": {
+          "oneOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "#/definitions/key"
+            }
+          ]
+        }
+      },
+      "required": [
+        "value",
+        "key"
+      ]
+    }
+  },
+  "type": "object",
+  "properties": {
+    "continue": {
+      "type": "object",
+      "properties": {
+        "uri": {
+          "type": "string"
+        },
+        "wait": {
+          "type": "integer"
+        },
+        "access_token": {
+          "$ref": "#/definitions/access_token",
+          "properties": {
+            "manage": false
+          }
+        }
+      },
+      "required": [
+        "uri"
+      ]
+    },
+    "access_token": {
+      "$ref": "#/definitions/access_token"
+    },
+    "multiple_access_tokens": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/access_token"
+      }
+    },
+    "interact": {
+      "type": "object",
+      "properties": {
+	    "redirect": {
+    	  "type": "string"
+      	},
+        "app": {
+          "type": "string"
+        },
+        "callback": {
+          "type": "string"
+        },
+        "user_code": {
+          "type": "object",
+          "properties": {
+            "code": { "type": "string" },
+            "url": { "type": "string" }
+          }
+        }
+      }
+    },
+    "subject": {
+      "type": [
+        "object",
+        "string"
+      ],
+      "properties": {
+        "sub_ids": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/subject_id_structures"
+          }
+        },
+        "assertions": {
+          "type": "object",
+          "propertyNames": {
+            "$ref": "#/definitions/assertion_types"
+          }/* no constraints on values right now*/
+        },
+        "updated_at": {
+          "type": "string",
+          "pattern": "([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2})\\:([0-9]{2})\\:([0-9]{2})"
```

Comment:
Not sure this pattern is needed, you should have an ISO8601 parser anyway.

Reply:
It's as much value as defining a string field as a `uri` type, since an application would have a URI parser also.

Reply:
Yes, but here you're using the schema definition to define a time format, maybe even a new time format.